File: Android/app/src/main/java/com/dreamproject/inwords/presentation/viewScenario/authorisation/login/LoginViewModel.java
package com.dreamproject.inwords.presentation.viewScenario.authorisation.login;

import com.dreamproject.inwords.data.dto.UserCredentials;
import com.dreamproject.inwords.domain.interactor.authorisation.AuthorisationInteractor;
import com.dreamproject.inwords.presentation.viewScenario.authorisation.AuthorisationViewModel;

import io.reactivex.Completable;

public class LoginViewModel extends AuthorisationViewModel {

    public LoginViewModel(AuthorisationInteractor authorisationInteractor) {
        super(authorisationInteractor);
    }

    @Override
    protected Completable performAuthAction(UserCredentials userCredentials) {
        return authorisationInteractor.signIn(userCredentials);
    }
}

File: Android/app/src/main/java/ru/inwords/inwords/presentation/view_scenario/authorisation/login/LoginViewModel.kt
package ru.inwords.inwords.presentation.view_scenario.authorisation.login

import io.reactivex.Completable
import ru.inwords.inwords.domain.interactor.authorisation.AuthorisationInteractor
import ru.inwords.inwords.presentation.view_scenario.authorisation.AuthorisationViewModel
import ru.inwords.inwords.profile.data.bean.UserCredentials

class LoginViewModel(authorisationInteractor: AuthorisationInteractor) : AuthorisationViewModel(authorisationInteractor) {
    override fun performAuthAction(userCredentials: UserCredentials): Completable {
        return authorisationInteractor.signIn(userCredentials)
    }
}